src/DependencyManagement/DependencyManager.cs

```diff
@@ -185,58 +188,86 @@ internal void Initialize(FunctionLoadRequest request)
                 }
+            }
 
-                try
+            try
+            {
+                foreach (DependencyInfo module in Dependencies)
                 {
-                    foreach (DependencyInfo module in Dependencies)
+                    string moduleName = module.Name;
+                    string latestVersion = module.LatestVersion;
+
+                    int tries = 1;
+
+                    while (true)
                     {
-                        string moduleName = module.Name;
-                        string latestVersion = module.LatestVersion;
-
-                        // Save the module to the given path
-                        pwsh.AddCommand("PowerShellGet\\Save-Module")
-                            .AddParameter("Repository", Repository)
-                            .AddParameter("Name", moduleName)
-                            .AddParameter("RequiredVersion", latestVersion)
-                            .AddParameter("Path", DependenciesPath)
-                            .AddParameter("Force", Utils.BoxedTrue)
-                            .AddParameter("ErrorAction", "Stop")
-                            .InvokeAndClearCommands();
-
-                        var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
-                        logger.Log(LogLevel.Trace, message, isUserLog: true);
+                        try
+                        {
+                            // Save the module to the given path
+                            pwsh.AddCommand("PowerShellGet\\Save-Module")
+                                .AddParameter("Repository", Repository)
+                                .AddParameter("Name", moduleName)
+                                .AddParameter("RequiredVersion", latestVersion)
+                                .AddParameter("Path", DependenciesPath)
+                                .AddParameter("Force", Utils.BoxedTrue)
+                                .AddParameter("ErrorAction", "Stop")
+                                .InvokeAndClearCommands();
+
+                            var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
+                            logger.Log(LogLevel.Trace, message, isUserLog: true);
+
+                            break;
+                        }
+                        catch (Exception e)
+                        {
+                            if (tries >= MaxNumberOfTries)
+                            {
+                                var errorMsg = string.Format(PowerShellWorkerStrings.FailToInstallFuncAppDependencies, e.Message);
+                                _dependencyError = new DependencyInstallationException(errorMsg, e);
+
+                                throw _dependencyError;
+                            }
+
+                            // Update the retry counter
+                            tries++;
```

Comment:
I think we should trace the errors in any case, even if we are going to retry.

Reply:
I agree. Fixed.